### DISCUSSION TITLE
Docker OpenWISP API error 'admin' is not a registered namespace

### DISCUSSION THREAD

**Discussion by 3pleo** [openwisp/docker-openwisp#360] [General]
I'm very new to openwisp and have just installed docker-openwisp and testing the API (Latest on Ubuntu 24.04)

When I try to create a user via the api, the user creation succeeds but throws error 'admin' is not a registered namespace with attached trace.

Is this a bug or am I doing something wrong?

Your guidance will be highly appreciated.
[docker_openwisp_api_error500_trace.txt](https://github.com/user-attachments/files/16944935/docker_openwisp_api_error500_trace.txt)

**Comment by pandafy** [comment]
Hello @3pleo, you have probably stumbled upon a bug in docker-openwisp. I have created a [PR to fix this issue in docker-openwisp](https://github.com/openwisp/docker-openwisp/pull/359). Thank you for reporting this. 

I assume that you are using the `edge` images for testing. In the meantime, you can add custom Django settings as workaround. 

Follow the steps mentioned in [Supplying Custom Django Settings](https://openwisp.io/docs/dev/docker/user/customization.html#id3). The content of `customization/configuration/django/custom_django_settings.py` should be the following: 

```python 
import os

REST_AUTH = {
    'SESSION_LOGIN': False,
    'PASSWORD_RESET_SERIALIZER': 'openwisp_radius.api.serializers.PasswordResetSerializer',
    'REGISTER_SERIALIZER': 'openwisp_radius.api.serializers.RegisterSerializer',
}

if os.environ['MODULE_NAME'] == 'api':
    LOGIN_REDIRECT_URL = 'account_change_password'
    ACCOUNT_LOGOUT_REDIRECT_URL = LOGIN_REDIRECT_URL
```

**Reply by 3pleo** [reply]
@pandafy

Thanks a million. This works.
When will the patch be available in edge images?

**Reply by pandafy** [reply]
You're welcome. The fix will be available in the `edge` images once https://github.com/openwisp/docker-openwisp/pull/359 is merged.

**Reply by nemesifier** [reply]
https://github.com/openwisp/docker-openwisp/pull/359 has been merged.

**Reply by 3pleo** [reply]
Thanks. The patch works.
But when I create a user via the api, the user gets email with a link to confirm email, which when clicked throws up the following similar error:

NoReverseMatch at /accounts/confirm-email/NTg:1sqSLG:FeIEip6aYPX77iN8vZ99FWKa5XWWvmRUlD2TxOK3qlY/
'admin' is not a registered namespace

find the full trace attached.
[confirm_email_error.txt](https://github.com/user-attachments/files/17027543/confirm_email_error.txt)

I noticed the following:
1. when I create user from the Dashboard, the link sent to email works perfectly and has the dashboard prefix url e.g. https://dashboard.mws.com/accounts/confirm-email/Njk:1sqYML:dWTn1rQBMbO9R4gHHmqgub8J7SF_U4JmzCzlQKBwKkc/
2. But when create user with REST API, "Confirm Email" url sent has the API prefix url e.g. https://api.mws.com/accounts/confirm-email/Njk:1sqYML:dWTn1rQBMbO9R4gHHmqgub8J7SF_U4JmzCzlQKBwKkc/. And throws up the not registered namespace error.

Kindly help.

Regards

**Reply by nemesifier** [reply]
Thanks for reporting this, we hadn't noticed this yet after the recent changes, this should fix it: https://github.com/openwisp/docker-openwisp/commit/efb2eb6e28466113a21429103f44cba92c864b41
